src/compile.rs

```diff
@@ -97,12 +97,20 @@ fn handle_module(
         context.func = func;
 
         // Compile and encode the result to machine code.
+        let total_size = context
+            .compile(isa)
+            .map_err(|err| pretty_error(&context.func, Some(isa), err))?;
+
         let mut mem = Vec::new();
+        mem.resize(total_size as usize, 0);
+
         let mut relocs = PrintRelocs { flag_print };
         let mut traps = PrintTraps { flag_print };
-        context
-            .compile_and_emit(isa, &mut mem, &mut relocs, &mut traps)
-            .map_err(|err| pretty_error(&context.func, Some(isa), err))?;
+        let mut code_sink: binemit::MemoryCodeSink;
+        unsafe {
+            code_sink = binemit::MemoryCodeSink::new(mem.as_mut_ptr(), &mut relocs, &mut traps);
+        }
```

Comment:
Why is it needed to call `compile` and `emit_function_to_memory` directly rather than using `compile_and_emit`? It's unfortunate to have `unsafe` here in user code, rather than having it be an implementation detail of the library.

Reply:
The reason I did this is because I needed access to the `MemoryCodeSink` for the `code_size` member, so I know how much to disassemble. In the `emit_to_memory` function of `context.rs` the `MemoryCodeSink` is created and thrown away. Since this was something only necessary for the CLI, I thought this code unlikely to be called by a user embedding and so not terribly important re: unsafe behavior. 

But if you think it should change, I can certainly make it work somehow.